code/game/objects/structures/crates_lockers/closets/bsdroppod.dm

```diff
@@ -0,0 +1,89 @@
+//The "BDPtarget" temp visual is created by the expressconsole, which in turn makes two things: a falling droppod animation, and the droppod itself.
+
+
+//------------------------------------BLUESPACE DROP POD-------------------------------------//
+/obj/structure/closet/bsdroppod
+	name = "Bluespace Drop Pod"
+	desc = "A Nanotrasen supply drop pod."
+	icon = 'icons/obj/2x2.dmi'
+	icon_state = "BDP"
+	pixel_x = -16//2x2 sprite
+	pixel_y = -5
+	pixel_z = 0
+	layer = TABLE_LAYER//so that the crate inside doesn't appear underneath
+	allow_objects = TRUE
+	allow_dense = TRUE
+	delivery_icon = null
+	can_weld_shut = FALSE
+	armor = list("melee" = 30, "bullet" = 50, "laser" = 50, "energy" = 100, "bomb" = 90, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 80)
+	anchored = TRUE
+	anchorable = FALSE
+	var/datum/supply_order/SupplyOrder
+
+/obj/structure/closet/bsdroppod/Initialize(mapload, datum/supply_order/so)
+	. = ..()
+	SupplyOrder = so//uses Supply Order passed from expressconsole into BDPtarget
+	addtimer(CALLBACK(src, .proc/open), 30)//open 3seconds after appearing
+
+/obj/structure/closet/bsdroppod/update_icon()//called in initialize, all we have to do is add a door icon
+	cut_overlays()//just incase update_icon somehow gets called multiple times, prevents overlay spam
+	add_overlay("BDP_door")
+
+/obj/structure/closet/bsdroppod/attackby(obj/item/W, mob/user, params)//i no need to be welded or anchored or whatever
+	return
+
+/obj/structure/closet/bsdroppod/toggle(mob/living/user)//ditto
+	return
+
+/obj/structure/closet/bsdroppod/open()
+	var/turf/T = get_turf(src)
+	SupplyOrder.generate(T)//not called during populateContents as supplyorder generation requires a turf
+	cut_overlays()
+	add_overlay("BDP_open")
+	playsound(src, open_sound, 15, 1, -3)
+	QDEL_IN(src, 30)
+
+/obj/structure/closet/bsdroppod/Destroy()//make some sparks b4 deletion
+	do_sparks(10, FALSE, src)
```

Comment:
add the do_sparks as a separate timer in Initialize

Reply:
i'll add it to open, as QDEL_IN is already there. Also, allows for manual-opening, if we feel like that for some reason down the line.